tapdb/universe_stats.go

```diff
@@ -575,14 +575,48 @@ func (u *UniverseStats) AggregateSyncStats(
 
 	log.Debugf("Populating aggregate sync stats")
 
-	dbStats, err := u.querySyncStats(ctx)
-	if err != nil {
+	var (
+		resChan = make(chan universe.AggregateStats, 1)
+		errChan = make(chan error, 1)
+		dbStats universe.AggregateStats
+	)
+
+	// We'll fire the db query in a separate go routine, with an undefined
+	// timeout. This way, we'll always retrieve the result regardless of
+	// what happens in the current context. This way we're always waiting
+	// for the call to complete and caching the result even if this
+	// function's result is an error.
+	go func() {
+		// Note: we have the statsMtx held, so even if a burst of
+		// requests took place in an un-cached state, this call would
+		// not be triggered multiple times.
+		dbStats, err := u.querySyncStats(context.Background())
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		log.Debugf("Retrieved aggregate sync stats: %+v", dbStats)
+
+		// We'll store the DB stats so that it can be read from cache
+		// later.
+		u.statsSnapshot.Store(&dbStats)
+
+		resChan <- dbStats
+	}()
+
+	select {
+	case <-ctx.Done():
+		log.Debugf("ctx timeout before retrieving aggregate sync stats")
```

Comment:
nit: Log statements should always start capitalized. Also, make it clear that the outer context timed out?